scripts/o3de/o3de/android.py

```diff
@@ -0,0 +1,608 @@
+
+        if args.list:
+            list_android_config(android_config)
+        if args.validate:
+            validate_android_config(android_config)
+        if args.set_value:
+            android_config.set_config_value_from_expression(args.set_value)
+        if args.set_password:
+            android_config.set_password(args.set_password)
+        if args.clear_value:
+            android_config.set_config_value(key=args.clear_value,
+                                            value='',
+                                            validate_value=False)
+            logger.info(f"Setting '{args.clear_value}' cleared.")
+
+    except (command_utils.O3DEConfigError, android_support.AndroidToolError) as err:
+        logger.error(str(err))
+        return 1
+    else:
+        return 0
+
+
+def prompt_validated_password(name: str) -> str:
+    """
+    Request a password with validation prompt to the user. If the password validation fails (either empty or doesnt match), an exception is thrown
+
+    :param name:    The password name to display
+    :return: The validated password
+    """
+    enter_password = getpass(f"Please enter the password for {name} : ")
+    if not enter_password:
+        raise android_support.AndroidToolError(f"Password for {name} required.")
+    confirm_password = getpass(f"Please verify the password for {name} : ")
+    if confirm_password != enter_password:
+        raise android_support.AndroidToolError(f"Passwords for {name} do not match.")
+
+    return enter_password
+
+
+def generate_android_project(args: argparse) -> int:
+    """
+    Execute the android gradle project creation workflow from the input arguments
+
+    :param args:    The args from the arg parser to extract the necessary parameters for generating the project
+    :return: The result code to return back
+
+    """
+
+    if args.debug:
+        logger.setLevel(logging.DEBUG)
+    else:
+        logger.setLevel(logging.INFO)
+
+    try:
+        # Create the android configuration
+        android_config, project_name = get_android_config_from_args(args)
+
+        # Resolve the project path and get the project and android settings
+        resolved_project_path = manifest.get_registered(project_name=project_name)
+        if not resolved_project_path:
+            raise android_support.AndroidToolError(f"Project '{project_name}' is not registered with O3DE.")
+        project_settings, android_settings = android_support.read_android_settings_for_project(resolved_project_path)
+
+        # Validate Java is installed
+        java_version = android_support.validate_java_environment()
+
+        # Validate gradle
+        gradle_path, gradle_version = android_support.validate_gradle(android_config)
+
+        # Validate cmake
+        cmake_path, cmake_version = android_support.validate_cmake(android_config)
+
+        # Validate ninja
+        ninja_path, ninja_version = android_support.validate_ninja(android_config)
```

Comment:
nbd, it almost seems like we could call `validate_android_config` for the validation logic.

Reply:
Fixed